src/main/java/ai/preferred/venom/Crawler.java

```diff
@@ -281,7 +281,9 @@ private void except(final Job job, final Throwable ex) {
       synchronized (pendingJobs) { // Synchronisation required to prevent crawler stopping incorrectly.
         pendingJobs.remove(job);
         if (job.getTryCount() < maxTries) {
-          job.reQueue();
+          job.prepareRetry();
+          queueScheduler.removeAndAdd(job);
```

Comment:
Why do we need remove followed by add? The job should not be in the queue already, just an add call should work well.